### ISSUE TITLE
argpars usage required args

### ISSUE THREAD

**Issue by dgud** [erlang/otp#7964]
Required arguments (options) are listed under **Optional Arguments**

I thought this was a bug and tried to solve that,
but then I discovered that the testcase test exactly that?

```
usage_required_args(Config) when is_list(Config) ->
    Cmd = #{commands => #{"test" => #{arguments => [#{name => required, required => true, long => "-req"}]}}},
    Expected = "Usage:\n  " ++ prog() ++ " test --req <required>\n\nOptional arguments:\n  --req required\n",
    ?assertEqual(Expected, unicode:characters_to_list(argparse:help(Cmd, #{command => ["test"]}))).
```

Shouldn't they be listed under arguments?

Any comments @max-au?

**Comment by max-au** [comment]
`argparse` has two types of arguments:
* positional (those not prefixed with `-`)
* optional (those prefixed with)

Parameter `-long` (or `-short`) is what makes the argument optional, hence `--req` is an _optional_ argument.

Now, marking it as _required_ does not change the argument type (it still stays optional). It only asks parser to throw an error when the option is not specified in the command line. This feature `#{required => true` is somewhat rare to use (and mainly needed to retrofit `argparse` for older software).

In other words, this isn't a bug, and works as expected.
